tesseract_task_composer/core/src/task_composer_node.cpp

```diff
@@ -52,7 +52,7 @@ TaskComposerNode::TaskComposerNode(std::string name, TaskComposerNodeType type,
 {
   try
   {
-    ns_ = config["namespace"] ? config["namespace"].as<std::string>() : name_;
+    ns_ = config["namespace"].IsDefined() ? config["namespace"].as<std::string>() : name_;
```

Comment:
Does this not work?
```
ns_ = (YAML::Node n = config["namespace"]) ? n.as<std::string>() : name_;
```

Reply:
It gave me a clang_tidy warning for that:
```bash
 /__w/tesseract_planning/tesseract_planning/target_ws/src/tesseract_task_composer/core/src/task_composer_node.cpp:55:11: error: implicit conversion 'YAML::detail::unspecified_bool_type' (aka 'void (*)(unspecified_bool::NOT_ALLOWED *)') -> bool [readability-implicit-bool-conversion,-warnings-as-errors]
    ns_ = config["namespace"] ? config["namespace"].as<std::string>() : name_;
          ^
                              != nullptr
```

Reply:
> Does this not work?
> 
> ```
> ns_ = (YAML::Node n = config["namespace"]) ? n.as<std::string>() : name_;
> ```

That doesn't compile, but I could potentially do:

```C++
YAML::Node n;
ns_ = (n = config["namespace"]) ? config["namespace"].as<std::string>() : name_;
```

Reply:
Okay, if CI passes then current state is good.